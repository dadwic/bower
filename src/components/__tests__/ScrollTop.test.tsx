import React from "react";
import renderer from "react-test-renderer";
import Fab from "@mui/material/Fab";
import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
import ScrollTop from "../ScrollTop";

describe("ScrollTop", () => {
  it("should corretly match snapshot", () => {
    const tree = renderer
      .create(
        <ScrollTop>
          <Fab color="secondary" size="small" aria-label="scroll back to top">
            <KeyboardArrowUpIcon />
          </Fab>
        </ScrollTop>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
