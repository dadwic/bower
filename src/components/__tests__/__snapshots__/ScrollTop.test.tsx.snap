// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollTop should corretly match snapshot 1`] = `
<div
  className="MuiBox-root css-mgur8"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "transform": "scale(0)",
      "visibility": "hidden",
    }
  }
>
  <button
    aria-label="scroll back to top"
    className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-secondary css-b7losd-MuiButtonBase-root-MuiFab-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="KeyboardArrowUpIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
      />
    </svg>
  </button>
</div>
`;
